src/components/ble/SimpleWeatherService.cpp

```diff
@@ -0,0 +1,151 @@
+}
+
+SimpleWeatherService::SimpleWeatherService(const DateTime& dateTimeController) : dateTimeController(dateTimeController) {
+}
+
+void SimpleWeatherService::Init() {
+  ble_gatts_count_cfg(serviceDefinition);
+  ble_gatts_add_svcs(serviceDefinition);
+}
+
+int SimpleWeatherService::OnCommand(struct ble_gatt_access_ctxt* ctxt) {
+  const auto* buffer = ctxt->om;
+  const auto* dataBuffer = buffer->om_data;
+
+  switch (GetMessageType(dataBuffer)) {
+    case MessageType::CurrentWeather:
+      if (GetVersion(dataBuffer) == 0) {
+        currentWeather = CreateCurrentWeather(dataBuffer);
+        NRF_LOG_INFO("Current weather :\n\tTimestamp : %d\n\tTemperature:%d\n\tMin:%d\n\tMax:%d\n\tIcon:%d\n\tLocation:%s",
+                     currentWeather->timestamp,
+                     currentWeather->temperature,
+                     currentWeather->minTemperature,
+                     currentWeather->maxTemperature,
+                     currentWeather->iconId,
+                     currentWeather->location);
+      }
+      break;
+    case MessageType::Forecast:
+      if (GetVersion(dataBuffer) == 0) {
+        forecast = CreateForecast(dataBuffer);
+        NRF_LOG_INFO("Forecast : Timestamp : %d", forecast->timestamp);
+        for (int i = 0; i < 5; i++) {
+          NRF_LOG_INFO("\t[%d] Min: %d - Max : %d - Icon : %d",
+                       i,
+                       forecast->days[i].minTemperature,
+                       forecast->days[i].maxTemperature,
+                       forecast->days[i].iconId);
+        }
+      }
+      break;
+    default:
+      break;
+  }
+
+  return 0;
+}
+
+std::optional<SimpleWeatherService::CurrentWeather> SimpleWeatherService::Current() const {
+  if (currentWeather) {
+    auto currentTime = dateTimeController.UTCDateTime().time_since_epoch();
+    auto weatherTpSecond = std::chrono::seconds {currentWeather->timestamp};
+    auto weatherTp = std::chrono::duration_cast<std::chrono::seconds>(weatherTpSecond);
+    auto delta = currentTime - weatherTp;
+
+    if (delta < std::chrono::hours {24}) {
+      return currentWeather;
+    }
+  }
+  return {};
+}
+
+std::optional<SimpleWeatherService::Forecast> SimpleWeatherService::GetForecast() const {
+  if (forecast) {
+    auto currentTime = dateTimeController.UTCDateTime().time_since_epoch();
+    auto weatherTpSecond = std::chrono::seconds {forecast->timestamp};
+    auto weatherTp = std::chrono::duration_cast<std::chrono::seconds>(weatherTpSecond);
+    auto delta = currentTime - weatherTp;
+
+    if (delta < std::chrono::hours {24}) {
+      return this->forecast;
+    }
+  }
+  return {};
+}
+
+bool SimpleWeatherService::CurrentWeather::operator==(const SimpleWeatherService::CurrentWeather& other) const {
+  return this->iconId == other.iconId && this->temperature == other.temperature && this->timestamp == other.timestamp &&
+         this->maxTemperature == other.maxTemperature && this->minTemperature == other.maxTemperature;
+}
```

Comment:
Shouldn't this also do a strcmp for the location?

Reply:
Good catch! Done!